ank/src/cli_commands/set_state.rs

```diff
@@ -117,9 +168,192 @@ impl CliCommands {
 //////////////////////////////////////////////////////////////////////////////
 #[cfg(test)]
 mod tests {
-    use std::io;
+    use super::*;
+    use crate::cli_commands::server_connection::MockServerConnection;
+    use common::{
+        commands::UpdateStateSuccess,
+        objects::{CompleteState, RestartPolicy, State},
+        state_manipulation::Object,
+    };
+    use mockall::predicate::eq;
+    use serde_yaml::Value;
+    use std::{collections::HashMap, io::Cursor};
+
+    pub async fn read_to_string_mock(_file: String) -> io::Result<String> {
+        Ok(_file.into())
+    }
+
+    const RESPONSE_TIMEOUT_MS: u64 = 3000;
+
+    const SAMPLE_CONFIG: &str = r#"desiredState:
+        workloads:
+          nginx:
+            agent: agent_A
+            tags:
+            - key: owner
+              value: Ankaios team
+            dependencies: {}
+            restartPolicy: ALWAYS
+            runtime: podman
+            runtimeConfig: |
+              image: docker.io/nginx:latest
+              commandOptions: ["-p", "8081:80"]"#;
+
+    #[test]
+    fn utest_add_default_workload_spec_empty_update_mask() {
+        let update_mask = vec![];
+        let mut complete_state = CompleteState {
+            desired_state: Default::default(),
+            ..Default::default()
+        };
+
+        add_default_workload_spec_per_update_mask(&update_mask, &mut complete_state);
+
+        assert!(complete_state.desired_state.workloads.is_empty());
+    }
+
+    #[test]
+    fn utest_add_default_workload_spec_with_update_mask() {
+        let update_mask = vec![
+            "desiredState.workloads.nginx.restartPolicy".to_string(),
+            "desiredState.workloads.nginx2.restartPolicy".to_string(),
+            "desiredState.workloads.nginx2".to_string(),
```

Comment:
I think it should be nginx3 instead of nginx2?
```suggestion
            "desiredState.workloads.nginx3".to_string(),
```